ui/utils.ts

```diff
@@ -125,6 +132,43 @@ export function setup_client_auditor() {
     }
 }
 
+let monitorMessages: Timer;
+
+export function setupMessageRetrieval() {
+
+    if (monitorMessages) {
+        return;
+    }
+
+    const toNotification = (message: string, severity: Severity, id: number): Notification => ({
+        title: '',
+        message: message,
+        severity: severity,
+        id: id
+    });
+
+    const consume = () => {
+        service.consumeMessages().then(value => {
+            let id = notificationManager.getNextId();
+            const errors = value.errors.map(error => {
+                return toNotification(error, Severity.ERROR, id++);
+            });
+            const warnings = value.warnings.map(warning => {
+                return toNotification(warning, Severity.WARNING, id++);
+            });
+
+            notificationManager.mergeToCache(errors.concat(warnings));
+            updateNotifications();
+            setupNotificationHandlers();
+        })
+            .catch(reason => {
+                console.error(reason);
+            });
+    };
+    consume();
+    monitorMessages = setInterval(consume, 15000);
```

Comment:
I guess every 10 seconds would even be better. It's a super cheap RPC call and the user should get these notification asap. Do you think 10 seconds is fine or even more frequent?

Reply:
If it doesn't cost much I don't see any reason why we can't go to 5 seconds.